crates/pathfinder/src/state/sync/l1.rs

```diff
@@ -15,25 +15,28 @@ pub async fn sync<T>(
     ethereum: T,
     chain: Chain,
     core_address: H160,
-) -> anyhow::Result<()>
-where
+) where
     T: EthereumApi + Clone,
 {
     let head_poll_interval = head_poll_interval(chain);
 
     loop {
-        let state_update = Retry::exponential(
+        let maybe_update = Retry::exponential(
             || async { ethereum.get_starknet_state(&core_address).await },
             NonZeroU64::new(1).unwrap(),
         )
         .factor(NonZeroU64::new(2).unwrap())
         .max_delay(head_poll_interval / 2)
         .when(|_| true)
-        .await?;
-
-        tx_event.send(state_update).await?;
-
+        .await;
+        match maybe_update {
+            Ok(state_update) => {
+                tx_event.send(state_update).await.ok();
+            }
+            Err(e) => {
+                tracing::warn!(reason=?e, "Failed to fetch L1 state");
```

Comment:
Using `.when(|_| true)` makes this branch unreachable :thinking:

Reply:
Tracing within the closure will probably be too much (each failure leading to retry will be logged). Maybe you want to cap the retries at some particular number with `max_num_retries(N)` which would make this branch log every Nth error (which is not ideal either) :shrug: .
Or maybe trace in the closure and warn once N failures, I can't make up my mind. You'll have a better idea :+1:

Reply:
Added a `deadline` retry mode. I think a few more lines in the code can explain better what is happening (retry behaviour). The deadline (half of the pull period) should do the trick.